### PR TITLE
Don't include docs in Array methods, just refer to module docs

### LINKED ISSUE #9223: Update dask.dataframe.map_partitions docs to match dask.dataframe.DataFrame.map_partitions
When I look up the documentation for `map_partitions`, I use the search box on the docs page. The top result is usually `dask.dataframe.map_partitions` so that is what I click on and that is the documentation I use as I assume that it is the same or similar to the docs for the method. Recently I wanted to see if map_partitions allows for a function argument like block_info. There is no mention of this on the docs page that I normally go to. I just went to `dask.dataframe.DataFrame.map_partitions` for the first time and I was really surprised by all of the extra information, including mention of a `partition_info` arg.

It seems like the module level function's docs are out of date compared to the `DataFrame` method's. It would be great if the docs were updated considering that the module level function's appear first in in-site searches.

### PR DESCRIPTION
**Pull request by jsignell**
- [ ] Closes #9223
- [x] Tests added / passed
- [x] Passes `pre-commit run --all-files`

~~I haven't quite finished looking for cases of this, and I'd like to find a way to enforce it, but this is what I have so far. I think I've found the main cases Array~~ This is good to go!

As a side benefit of this approach now the Array summary is much more interesting

Before
image

After
image

### CODE REVIEW

File: dask/array/core.py [hunk 1]
```diff
@@ -2376,48 +2412,98 @@ def __rdivmod__(self, other):
 
         return divmod(other, self)
 
-    @derived_from(np.ndarray)
     def any(self, axis=None, keepdims=False, split_every=None, out=None):
+        """Returns True if any of the elements of `a` evaluate to True.
```
Comment (jrbourbeau) [inline dask/array/core.py:2416]:
```suggestion
        """Returns True if any of the elements of ``a`` evaluate to True.
```

File: dask/array/core.py [hunk 2]
```diff
@@ -2165,8 +2198,11 @@ def topk(self, k, axis=-1, split_every=None):
     def argtopk(self, k, axis=-1, split_every=None):
         """The indices of the top k elements of an array.
 
-        See :func:`dask.array.argtopk` for docstring.
+        Refer to :func:`dask.array.argtopk` for full doctumentation.
```
Comment (jrbourbeau) [inline dask/array/core.py:2201]:
```suggestion
        Refer to :func:`dask.array.argtopk` for full documentation.
```

File: dask/array/core.py [hunk 3]
```diff
@@ -2557,136 +2652,68 @@ def map_blocks(self, func, *args, **kwargs):
...
+        --------
+        dask.array.cumprod : equivalent function
         """
         from dask.array.reductions import cumprod
 
         return cumprod(self, axis, dtype, out=out, method=method)
 
-    @derived_from(np.ndarray)
     def squeeze(self, axis=None):
+        """Remove axes of length one from `a`.
```
Comment (jrbourbeau) [inline dask/array/core.py:2694]:
```suggestion
        """Remove axes of length one from ``a``.
```

File: dask/array/core.py [hunk 4]
```diff
@@ -2757,14 +2800,28 @@ def view(self, dtype=None, order="C"):
             chunk.view, dtype, order=order, dtype=dtype, chunks=chunks
         )
 
-    @derived_from(np.ndarray)
     def swapaxes(self, axis1, axis2):
+        """Return a view of the array with `axis1` and `axis2` interchanged.
```
Comment (jrbourbeau) [inline dask/array/core.py:2804]:
```suggestion
        """Return a view of the array with ``axis1`` and ``axis2`` interchanged.
```

File: dask/array/core.py (hunk 3 above)
Comment (jrbourbeau) [reply dask/array/core.py:2694]: Actually, we should use `self` instead of `a`

File: dask/array/core.py [hunk 5]
```diff
@@ -2757,14 +2800,28 @@ def view(self, dtype=None, order="C"):
...
+        --------
+        dask.array.swapaxes : equivalent function
+        """
         from dask.array.routines import swapaxes
 
         return swapaxes(self, axis1, axis2)
 
-    @derived_from(np.ndarray)
     def round(self, decimals=0):
+        """Return `a` with each element rounded to the given number of decimals.
```
Comment (jrbourbeau) [inline dask/array/core.py:2817]:
```suggestion
        """Return `a` with each element rounded to the given number of decimals.
```

Comment (jrbourbeau) [reply dask/array/core.py:2817]: Actually, we should use `self` instead of `a`

**Review by jrbourbeau — COMMENTED** [review]
Thanks @jsignell! Overall this looks good -- just left a few minor comments. While this PR is useful, I don't think it actually resolves https://github.com/dask/dask/issues/9223 as that issue was specifically concerned with `dask.dataframe.map_partitions` / `dask.DataFrame.map_partitions`. 

Also, FWIW, I like your idea of using a decorator for this (xref https://github.com/dask/dask/issues/9223#issuecomment-1169208059). Not saying we need to include it in this PR, but it would be nice to have a separate issue for that so we don't loose track of the idea

**Comment by jsignell** [comment]
> I don't think it actually resolves #9223 as that issue was specifically concerned with `dask.dataframe.map_partitions` / `dask.DataFrame.map_partitions`.

:facepalm: good point. I will do this for dataframe.core as well.

File: dask/array/core.py (hunk 3 above)
Comment (jsignell) [reply dask/array/core.py:2694]: I changed it to `from array` I prefer that to the other options.

File: dask/array/core.py (hunk 5 above)
Comment (jrbourbeau) [reply dask/array/core.py:2817]: Nice suggestion -- I think `array` is more clear 👍

Comment (jsignell) [reply dask/array/core.py:2817]:
```suggestion
        """Return array with each element rounded to the given number of decimals.
```

**Pull request merged**